src/SoftRobots/component/constraint/model/CableModel.cpp

```diff
@@ -41,12 +40,158 @@ namespace component
+unsigned int CableModel<Vec2Types>::computeClosestIndice(Coord position)
+{
+    double closest_distance = (position - Coord(m_topology->getPX(0), m_topology->getPY(0))).norm();
+    unsigned int closest_vertice = 0;
+    for(unsigned int j=1; j<m_topology->getNbPoints(); j++)
+    {
+        double distance = (position - Coord(m_topology->getPX(j), m_topology->getPY(j))).norm();
+        if(distance < closest_distance) 
+        {
+            closest_distance = distance;
+            closest_vertice = j;
+        }
+    } 
+   return closest_vertice;
+}
+
+
+template<>
+void CableModel<Vec2Types>::computePointsActionArea()
+{
+    // Implementing continuous Dijkstra as a geodesic distance measure
+    ReadAccessor<Data<VecCoord>> cablePositions = m_state->readPositions();
+    unsigned int nbCenters = d_indices.getValue().size();
+    const SetIndexArray &indices = d_indices.getValue();
+    ReadAccessor<Data<vector<Real>>> maxDistance = d_radii;
+    vector<Real> m_totalRatio;
+    m_areaIndices.clear();
+    m_areaIndices.resize(nbCenters);
+    m_ratios.clear();
+    m_ratios.resize(nbCenters);
+    m_totalRatio.clear();
+    m_totalRatio.resize(nbCenters);
+
+    // Find closest indice on surface topology
+    SetIndexArray m_closestCenterIndices;
+    if (nbCenters != 0)
+    {
+        m_closestCenterIndices.resize(nbCenters);
+        for(unsigned int i=0; i<nbCenters; i++)
+            m_closestCenterIndices[i] = computeClosestIndice(cablePositions[indices[i]]);
+    }
+
+    // Iterate for each cable attachment
+    typedef pair<Real,BaseMeshTopology::PointID> DistanceToPoint;
+    for (unsigned int i=0; i<nbCenters; i++)
+    {   
+        // Init
+        set<DistanceToPoint> queue; 
+        typename set<DistanceToPoint>::iterator qit;
+        vector<Real> distances(m_topology->getNbPoints(),maxDistance[i]);
+
+        //////////////////// If cable attachment does not match with a mesh vertice /////////////////
+        //////////// Project on closest triangle and distribute distance to its vertices ////////////
+        const TrianglesAroundVertex& trianglesAroundClosestVertex = m_topology->getTrianglesAroundVertex(m_closestCenterIndices[i]); // Triangles connected to closest vertice
+        static sofa::helper::DistancePointTri proximitySolver;
+        Vector3 projectionOnTriangle;
+        Real minDistanceToTriangle = std::numeric_limits<Real>::max(); unsigned int closestTriangleId = 0; Vector2 closestProjectionOnTriangle;
+        for(unsigned int j=0; j<trianglesAroundClosestVertex.size(); j++)
+        {
+            const Triangle triangle =  m_topology->getTriangle(trianglesAroundClosestVertex[j]);   
+            Vector3 p0 = Vector3(m_topology->getPX(triangle[0]), m_topology->getPY(triangle[0]), 0.0);
+            Vector3 p1 = Vector3(m_topology->getPX(triangle[1]), m_topology->getPY(triangle[1]), 0.0);
+            Vector3 p2 = Vector3(m_topology->getPX(triangle[2]), m_topology->getPY(triangle[2]), 0.0);           
```

Comment:
```suggestion
            Vector3 p0 {m_topology->getPX(triangle[0]), m_topology->getPY(triangle[0]), 0.0};
            Vector3 p1 {m_topology->getPX(triangle[1]), m_topology->getPY(triangle[1]), 0.0};
            Vector3 p2 {m_topology->getPX(triangle[2]), m_topology->getPY(triangle[2]), 0.0};           
```
and I bet they can be const